CRM/Utils/String.php

```diff
@@ -1050,7 +1075,10 @@ public static function parseOneOffStringThroughSmarty($templateString, $template
       // is invalid in Windows, causing failure.
       // Adding this is preparatory to smarty 3. The original PR failed some
       // tests so we check for the function.
-      if (!function_exists('smarty_function_eval') && (!defined('SMARTY_DIR') || !file_exists(SMARTY_DIR . '/plugins/function.eval.php'))) {
+      $canUseEvalResourceType = !function_exists('smarty_function_eval')
+        && (!defined('SMARTY_DIR')
+          || !file_exists(SMARTY_DIR . '/plugins/function.eval.php'));
+      if (!$useCache && $canUseEvalResourceType) {
```

Comment:
my only concern here is the windows element here @demeritcowboy do you have thoughts here?

Reply:
Something else seems to have changed since 5.69. Let me get back to you...

Reply:
Just to be clear, this is a very simple refactoring. The function `parseOneOffStringThroughSmarty()` should work exactly as before. So any problems with Windows won't have changed.

Reply:
Yes the old function might be unchanged but the question is about the new function/usecache=true. It will try to use `string:`, and this is where the problem is (was). I'm in and out today but I'd like to (a) determine if there is still a problem, (b) understand what changed elsewhere, and (c) remember what (c) was.